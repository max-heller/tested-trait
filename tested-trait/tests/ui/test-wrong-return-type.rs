use tested_trait::tested_trait;

fn main() {
    #[tested_trait]
    trait Foo {
        #[test]
        fn test() {
            5
        }

        #[test]
        fn test2() -> Result<(), ()> {}
    }
}
