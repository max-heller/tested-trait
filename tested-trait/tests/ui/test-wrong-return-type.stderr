error[E0308]: mismatched types
 --> tests/ui/test-wrong-return-type.rs:8:13
  |
8 |             5
  |             ^ expected `()`, found integer

error[E0308]: mismatched types
  --> tests/ui/test-wrong-return-type.rs:12:38
   |
12 |         fn test2() -> Result<(), ()> {}
   |                                      ^^ expected `Result<(), ()>`, found `()`
   |
   = note:   expected enum `Result<(), ()>`
           found unit type `()`
